src/provider/provider_os_memory.c

```diff
@@ -237,6 +233,8 @@ void os_finalize(void *provider) {
     free(os_provider);
 }
 
+umf_result_t os_get_min_page_size(void *provider, void *ptr, size_t *page_size);
```

Comment:
not related to this PR, but why we are marking only the `os_alloc` function as `static`?

nit: Instead of forward declaration we can just move `os_get_min_page_size` function definition here.

Reply:
> not related to this PR, but why we are marking only the `os_alloc` function as `static`?
> 
Done: https://github.com/oneapi-src/unified-memory-framework/pull/92